lib2/Ppmd8Tdecoder.c

```diff
@@ -0,0 +1,138 @@
+//
+// Created by miurahr on 2021/08/07.
+//
+
+#include "Ppmd8Tdecoder.h"
+
+PPMD_pthread_mutex_t mutex;
+PPMD_pthread_cond_t finished, inEmpty, notEmpty;
+
+Byte TReader(const void *p) {
+    BufferReader *bufferReader = (BufferReader *)p;
+    if (bufferReader->inBuffer->pos == bufferReader->inBuffer->size) {
+        PPMD_pthread_cond_signal(&inEmpty);
```

Comment:
This should be called under mutex locked.
L54, and L56 places lock and unlock then here is locked.
It is not obvious conditions, so it is better to notice it in comment.